e2e_tests/tests/task/test_generic_tasks.py

```diff
@@ -0,0 +1,281 @@
+        context_directory = context.read_v1_context(
+            pathlib.Path(conf.fixtures_path("generic_task")), []
+        )
+
+        req = bindings.v1CreateGenericTaskRequest(
+            config=config_text,
+            contextDirectory=context_directory,
+            projectId=None,
+            forkedFrom=None,
+            parentId=None,
+            inheritContext=False,
+            noPause=False,
+        )
+        task_resp = bindings.post_CreateGenericTask(test_session, body=req)
+
+    # Create fork task
+    with open(conf.fixtures_path("generic_task/test_config_fork.yaml"), "r") as fork_config_file:
+        config = ntsc.parse_config(fork_config_file, None, [], [])
+        config_text = util.yaml_safe_dump(config)
+        context_directory = context.read_v1_context(
+            pathlib.Path(conf.fixtures_path("generic_task")), []
+        )
+
+        req = bindings.v1CreateGenericTaskRequest(
+            config=config_text,
+            contextDirectory=context_directory,
+            projectId=None,
+            forkedFrom=task_resp.taskId,
+            parentId=None,
+            inheritContext=False,
+            noPause=False,
+        )
+        fork_task_resp = bindings.post_CreateGenericTask(test_session, body=req)
+
+        # Get fork task Config
+        command = ["det", "-m", conf.make_master_url(), "task", "config", fork_task_resp.taskId]
+
+        res = subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+        result_config = util.yaml_safe_load(res.stdout)
+        expected_config = {"entrypoint": ["echo", "forked"]}
+        assert expected_config == result_config
+
+
+@pytest.mark.e2e_cpu
+def test_kill_generic_task() -> None:
+    """
+    Start a simple task without a context directory and grab its config
+    """
+    test_session = api_utils.determined_test_session()
+
+    with open(conf.fixtures_path("generic_task/test_config.yaml"), "r") as config_file:
+        # Create task
+        config = ntsc.parse_config(config_file, None, [], [])
+        config_text = util.yaml_safe_dump(config)
+        context_directory = context.read_v1_context(
+            pathlib.Path(conf.fixtures_path("generic_task")), []
+        )
+
+        req = bindings.v1CreateGenericTaskRequest(
+            config=config_text,
+            contextDirectory=context_directory,
+            projectId=None,
+            forkedFrom=None,
+            parentId=None,
+            inheritContext=False,
+            noPause=False,
+        )
+        task_resp = bindings.post_CreateGenericTask(test_session, body=req)
+
+        # Kill task
+        command = ["det", "-m", conf.make_master_url(), "task", "kill", task_resp.taskId]
+
+        subprocess.run(command, universal_newlines=True, stdout=subprocess.PIPE, check=True)
+
+        kill_resp = bindings.get_GetTask(test_session, taskId=task_resp.taskId)
+        assert bindings.v1GenericTaskState.CANCELED == kill_resp.task.taskState
```

Comment:
Nit.

This is known as a "yoda check" because it reads like "make sure canceled the task is".

Can you reorder to be `assert <value-under-test> == <expected-value>`?  That is a far more common idiom that will make your code more easily readable by other developers.

This also applies to other places in your PR.